### PR TITLE
feat: Create Geneva docker compose files

### LINKED ISSUE #282: Create Geneva compose files
Create `geneva` folder under releases
Copy nightly build compose files over to `geneva` folder and rename appropriately
Replace all instances of `nexus3.edgexfoundry.org:10004` with `edgexfoundry`
Replace the `master` tag with the appropriate version number.
    - 1.2.0 for all edgex-go services, consul, mongo & support-rulesengine
    - 1.1.0 for app-service-configurable, device-rest, device-mqtt-go
    - UI and other device service TBD

### PR DESCRIPTION
**Pull request by lenny-intel**
Created Geneva docker compose files
Add comments for deprecating Support Rules Engine and Mongo

closes #282 & #272

### CODE REVIEW

**Comment by bnevis-i** [comment]
Script to aid in review:

```
diff ../nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml docker-compose-geneva-mongo-arm64.yml
diff ../nightly-build/compose-files/docker-compose-nexus-mongo-no-secty-arm64.yml docker-compose-geneva-mongo-no-secty-arm64.yml
diff ../nightly-build/compose-files/docker-compose-nexus-mongo-no-secty.yml docker-compose-geneva-mongo-no-secty.yml
diff ../nightly-build/compose-files/docker-compose-nexus-mongo.yml docker-compose-geneva-mongo.yml
diff ../nightly-build/compose-files/docker-compose-nexus-redis-arm64.yml docker-compose-geneva-redis-arm64.yml
diff ../nightly-build/compose-files/docker-compose-nexus-redis-no-secty-arm64.yml docker-compose-geneva-redis-no-secty-arm64.yml
diff ../nightly-build/compose-files/docker-compose-nexus-redis-no-secty.yml docker-compose-geneva-redis-no-secty.yml
diff ../nightly-build/compose-files/docker-compose-nexus-redis.yml docker-compose-geneva-redis.yml
diff ../nightly-build/compose-files/docker-compose-portainer.yml docker-compose-portainer.yml
```

**Comment by lenny-intel** [comment]
> Script to aid in review:

NICE!! THX!

**Comment by bnevis-i** [comment]
I've noticed that while most of the containers are version 1.2.0, some are 1.1.0:

```
docker-compose-geneva-mongo-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable-arm64:1.1.0
docker-compose-geneva-mongo-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go-arm64:1.1.0
docker-compose-geneva-mongo-no-secty-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable-arm64:1.1.0
docker-compose-geneva-mongo-no-secty-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go-arm64:1.1.0
docker-compose-geneva-mongo-no-secty.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable:1.1.0
docker-compose-geneva-mongo-no-secty.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go:1.1.0
docker-compose-geneva-mongo.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable:1.1.0
docker-compose-geneva-mongo.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go:1.1.0
docker-compose-geneva-redis-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable-arm64:1.1.0
docker-compose-geneva-redis-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go-arm64:1.1.0
docker-compose-geneva-redis-no-secty-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable-arm64:1.1.0
docker-compose-geneva-redis-no-secty-arm64.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go-arm64:1.1.0
docker-compose-geneva-redis-no-secty.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable:1.1.0
docker-compose-geneva-redis-no-secty.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go:1.1.0
docker-compose-geneva-redis.yml:    image: nexus3.edgexfoundry.org:10004/docker-app-service-configurable:1.1.0
docker-compose-geneva-redis.yml:    image: nexus3.edgexfoundry.org:10004/docker-device-rest-go:1.1.0
```

**Comment by lenny-intel** [comment]
> I've noticed that while most of the containers are version 1.2.0, some are 1.1.0:

Correct, some services like App Service Configurable & REST DS are 1.1.0. App Services and Device Service have their own release series.

**Comment by lenny-intel** [comment]
> Gave a thorough diff inspection w.r.t. nightlies. All the differences I found were explainable. Still could use someone to go through and smoke test it all.

Excellent! Jim and I are working on the smoke tests of the non-secure X86 versions now.

File: releases/geneva/docker-compose-geneva-redis-no-secty-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+      - consul
+#      - logging  # uncomment if re-enabled remote logging
+      - redis
+      - notifications
+
+  data:
+    image: edgexfoundry/docker-core-data-go-arm64:1.2.0
+    ports:
+      - "127.0.0.1:48080:48080"
+      - "127.0.0.1:5563:5563"
```
[the remark this answers is not in the document]
Comment (bnevis-i) [reply releases/geneva/docker-compose-geneva-redis-no-secty-arm64.yml:166]: That's the zero MQ port.

File: releases/geneva/docker-compose-geneva-redis-arm64.yml [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+      - /tmp/edgex/secrets/edgex-security-proxy-setup:/tmp/edgex/secrets/edgex-security-proxy-setup:ro,z
+    depends_on:
+      - consul
+      - vault-worker
+      - kong
+
+# end of containers for reverse proxy
+
+  redis:
+    image: redis:5.0.8-alpine
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply releases/geneva/docker-compose-geneva-redis-arm64.yml]: Fixed

File: releases/geneva/docker-compose-geneva-redis-no-secty-arm64.yml [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+    depends_on:
+      - consul
+#      - logging  # uncomment if re-enabled remote logging
+      - data
+
+  rulesengine:
+    image: emqx/kuiper:0.3.2
+    ports:
+      - "127.0.0.1:48075:48075"
+      - "127.0.0.1:20498:20498"
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply releases/geneva/docker-compose-geneva-redis-no-secty-arm64.yml:241]: Yes. kuiper set this up.

**Pull request merged**